packages/cdktf/lib/terraform-element.ts

```diff
@@ -17,8 +19,14 @@ export class TerraformElement extends Construct {
    */
   private _logicalIdOverride?: string;
 
-  constructor(scope: Construct, id: string) {
+  /**
+   * Type of this element, used for fqn
+   */
```

Comment:
I'm, wondering whether we should add a note why the `elementType` may be undefined. If I saw that correctly, that was for example the case for `Provider` constructs, right?

Reply:
I added a bit more info in the comment, please check it out

Reply:
looks good 👍